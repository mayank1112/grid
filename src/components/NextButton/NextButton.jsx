import PropTypes from 'prop-types';
import { memo } from 'react';
import { Button } from '@material-ui/core';

const NextButton = ({ onClick, disabled }) => {
    const customProps = disabled ? { disabled: '' } : {};
    return (
        <div className='button nextButton'>
            <Button {...customProps} onClick={onClick} variant="contained" color="primary">Next</Button>
        </div>
    );
};

NextButton.propTypes = {
    onClick: PropTypes.func.isRequired,
    disabled: PropTypes.bool.isRequired,
};

export default memo(NextButton);
