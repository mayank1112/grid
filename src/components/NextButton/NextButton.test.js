import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import NextButton from './NextButton';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });
const mockClick = jest.fn();

it('<NextButton> component should match snapshot when button is enabled', () => {
    const wrapper = shallow(<NextButton onClick={mockClick} disabled={false} />);
    expect(toJson(wrapper)).toMatchSnapshot();
});

it('<NextButton> component should match snapshot when button is disabled', () => {
    const wrapper = shallow(<NextButton onClick={mockClick} disabled={true} />);
    expect(toJson(wrapper)).toMatchSnapshot();
});

it('<NextButton> should not call onClick function on click of disabled button', () => {
    const wrapper = mount(<NextButton onClick={mockClick} disabled={true} />);
    const button = wrapper.find('.nextButton');
    button.simulate('click');
    expect(mockClick).not.toHaveBeenCalled();
});


it('<NextButton> should call onClick function on click of button', () => {
    const wrapper = mount(<NextButton onClick={mockClick} disabled={false} />);
    const button = wrapper.find('button');
    button.simulate('click');
    expect(mockClick).toHaveBeenCalledTimes(1);
});
