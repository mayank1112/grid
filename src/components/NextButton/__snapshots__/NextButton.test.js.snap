// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NextButton> component should match snapshot when button is disabled 1`] = `
<div
  className="button nextButton"
>
  <WithStyles(ForwardRef(Button))
    color="primary"
    disabled=""
    onClick={[MockFunction]}
    variant="contained"
  >
    Next
  </WithStyles(ForwardRef(Button))>
</div>
`;

exports[`<NextButton> component should match snapshot when button is enabled 1`] = `
<div
  className="button nextButton"
>
  <WithStyles(ForwardRef(Button))
    color="primary"
    onClick={[MockFunction]}
    variant="contained"
  >
    Next
  </WithStyles(ForwardRef(Button))>
</div>
`;
